sparkle/src/components/SplitButton.tsx

```diff
@@ -0,0 +1,87 @@
+import { cva } from "class-variance-authority";
+import React, { useState } from "react";
+
+import {
+  Button,
+  ButtonSizeType,
+  MetaButtonProps,
+} from "@sparkle/components/Button";
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+} from "@sparkle/components/Dropdown";
+import { Separator } from "@sparkle/components/Separator";
+import { ChevronDownIcon } from "@sparkle/icons";
+import { cn } from "@sparkle/lib";
+
+const separatorSizeVariants: Record<ButtonSizeType, string> = {
+  xs: "s-h-3",
+  sm: "s-h-5",
+  md: "s-h-7",
+};
+
+const separatorVariants = cva("s--ml-[1px]", {
+  variants: {
+    size: separatorSizeVariants,
+  },
+});
+type LabelProps = {
+  label: string;
+  icon?: React.ComponentType;
+  tooltip?: string;
+  disabled?: boolean;
+  onClick?: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
+};
+type SplitButtonProps = MetaButtonProps & {
+  values: LabelProps[];
+};
+
+export const SplitButton = React.forwardRef<
+  HTMLButtonElement,
+  SplitButtonProps
+>(({ values, className, size, variant, disabled, ...props }, ref) => {
+  const [current, setCurrent] = useState(values[0]);
```

Comment:
NIT: I wonder if it would be easier to use if the current was set with an attribute like 'active' or 'current' in LabelProps, and the others are displayed by order in the array. It would avoid having us reordering the list constantly? WDYT? Not strongly held!